api/lib/domain/usecases/finalize-session.js

```diff
@@ -43,3 +63,43 @@ module.exports = async function finalizeSession({
     sessionTime: finalizedSession.time,
   });
 };
+
+function _hasAbortReasonForCompletedCertificationCourse({ abortReasonCount, unfinishedCertificationCount }) {
+  return abortReasonCount > unfinishedCertificationCount;
+}
+
+function _countAbortReasons(certificationReports) {
+  let abortReasonCount = 0;
+  certificationReports.forEach(({ abortReason }) => {
+    if (abortReason) {
+      abortReasonCount += 1;
+    }
+  });
+
+  return abortReasonCount;
```

Comment:
On peut utiliser plus simplement un filter ou un reduce:
```suggestion
let abortReasonCount = 0;
  const abortReasonCount =  certificationReports.filter(({ abortReason }) => abortReason).length
  return abortReasonCount;
```